engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/optimise/LambdaConsolidate.scala

```diff
@@ -392,16 +394,23 @@ case object LambdaConsolidate extends IRPass {
         // Empty set is used to indicate that it isn't shadowed
         val usageIds =
           if (isShadowed) {
-            aliasInfo.graph
+            val occurs = aliasInfo.graph
               .linksFor(aliasInfo.id)
+              .stream
               .filter(_.target == aliasInfo.id)
               .map(link => aliasInfo.graph.getOccurrence(link.source))
-              .collect {
+              .map {
                 case Some(
                       GraphOccurrence.Use(_, _, identifier, _)
                     ) =>
-                  identifier
+                  Some(identifier)
+                case _ => None
               }
+              .filter(_.isDefined)
+              .map(_.get)
```

Comment:
This is wrong, it should just be a `flatMap`

Reply:
It is `java.util.stream.Stream` - it doesn't understand `scala.Option`. Does your comment still apply even it is a Java stream?